qemu/tests/blockdev_commit_standby.py

```diff
@@ -0,0 +1,40 @@
+from provider import backup_utils
+from provider import job_utils
+
+from provider.blockdev_commit_base import BlockDevCommitTest
+
+
+class BlockdevCommitStandby(BlockDevCommitTest):
+    def commit_snapshots(self):
+        device = self.params.get("device_tag")
+        device_params = self.params.object_params(device)
+        snapshot_tags = device_params["snapshot_tags"].split()
+        self.device_node = self.get_node_name(device)
+        device = self.get_node_name(snapshot_tags[-1])
+        commit_cmd = backup_utils.block_commit_qmp_cmd
+        cmd, args = commit_cmd(device)
+        self.main_vm.monitor.cmd(cmd, args)
+        job_id = args.get("job-id", device)
+        job_utils.wait_until_job_status_match(self.main_vm, 'ready', job_id, timeout=120)
+        self.main_vm.monitor.cmd("job-pause", {'id': job_id})
+        if not job_utils.is_block_job_paused(self.main_vm, job_id):
+            self.test.fail("Block job is not paused")
+        self.main_vm.monitor.cmd("job-complete", {'id': job_id})
+        self.main_vm.monitor.cmd("job-resume", {'id': job_id})
+        job_utils.wait_until_block_job_completed(self.main_vm, job_id)
```

Comment:
@aliang123 The 2nd time 'job-complete' would be executed in the function. It seems a redundancy.

Reply:
@ngu-niny, have updated, please help to check again, thanks.

Reply:
@aliang123 It's better to move the "job-complete" command to job_utils.wait_until_block_job_completed() function too.

Reply:
@ngu-niny, have updated. Please help to check again.

Reply:
I am afraid we cannot move all operations into the wait_until_block_job_completed function, which is widely used, the change might cause regressions. what about:

self.main_vm.monitor.cmd("job-pause", {'id': job_id})
job_utils.block_job_complete()
self.main_vm.monitor.cmd("job-resume", {'id': job_id})
if job_utils.get_event_by_condition("BLOCK_JOB_COMPLETED") is None:
case fail.

This explicitly follow your polarion test steps, and no code change in job_utils.py
Finally, if we wanna check the job is removed, we can use
job_utils.wait_until_job_status_match("null")

Reply:
zhenchao, after self.main_vm.monitor.cmd("job-pause", {'id': job_id}), job status reach 'standby', but in job_utils.block_job_complete(), we need to match job status 'ready'. So vm.monitor.cmd("job-complete", {'id': job_id}) needed here.

And "job_utils.get_event_by_condition("BLOCK_JOB_COMPLETED") is None" is a good solution, I think.

Reply:
OK, right, let's send the job-complete command directly.